Dockerfile

```diff
@@ -116,8 +116,8 @@ ARG TARGETARCH
 # https://www.docker.com/blog/faster-multi-platform-builds-dockerfile-cross-compilation-guide/
 RUN --mount=type=cache,target=/root/.cache/go-build \
     --mount=type=cache,target=/go/pkg \
-    GOOS=${TARGETOS:-linux} \
-    GOARCH=${TARGETARCH:-amd64} \
+    export GOOS=${TARGETOS:-linux} && \
```

Comment:
@tjohnson31415 -- I know I missed the review window on this one, but would it not have been more _Docker-esque_ to pull the two variables out into a Dockerfile `ENV` declaration?

```Dockerfile
ENV GOOS=${TARGETOS:-linux} \
    GOARCH=${TARGETARCH:-amd64}
RUN --mount=... \
    go build -o ...
```

Only `RUN`, `COPY`, and `ADD` create layers

Reply:
Well, these two environment variables are configurations only for the single RUN command. If this docker stage was `FROM`'d in a final pushed image, I'd think it better avoid the extra env vars created by `ENV`, but this is an intermediate stage so 🤷